### PR TITLE
Obmin 291 portal

### PR DESCRIPTION
**Pull request by Viktoria-Serkiz**
Created modal component

### CODE REVIEW

File: src/components/modal/styles.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  100% {
+    opacity: 1;
+  }
+`;
+
+const fadeOut = keyframes`
+ 0% {
+    opacity: 1;
+  }
+  100%{
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:19]: space

File: src/components/modal/styles.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    opacity: 0;
+  }
+`;
+
+const continueMoveDown = keyframes`
+ 0% {
+   opacity: 1;
+   transform: translateY(0vh);
+  }
+  100%{
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:29]: space

File: src/components/modal/styles.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+     transform: translateY(100vh);
+  }
+`;
+
+const moveDown = keyframes`
+ 0% {
+   opacity: 0;
+     
+  }
+  5%{
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:40]: space

File: src/components/modal/styles.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  100%{
+    opacity: 0;
+     transform: translateY(100vh);
+  }
+`;
+
+const moveDown = keyframes`
+ 0% {
+   opacity: 0;
+
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:38]: delete space

File: src/components/modal/styles.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+const moveDown = keyframes`
+ 0% {
+   opacity: 0;
+     
+  }
+  5%{
+    transform: translateY(-100vh);
+  }
+  100%{
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:43]: space

File: src/components/modal/styles.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 30px;
+  height: 30px;
+  top: -15px;
+  right: -15px;
+  border-radius: 50%;
+  background-color: ${({ theme }) =>
+    theme.colors.modalColors.crossBtnBackground};
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:93]: if you have 2 callbacks
you need to do once

File: src/config/theme/colors.ts [hunk 7]
```diff
@@ -136,10 +139,17 @@ const colorsDefault = Object.freeze({
...
+  background: modalBackground,
+  cross: crossBtn,
+  crossBtnBackground: crossBtnBackground,
+});
+
 const COLORS = Object.freeze({
   input,
   button,
   itemCard,
+  modalColors,
```
Comment (Wolshebnik) [inline src/config/theme/colors.ts:152]: sort

**Review by Mish-hash — APPROVED** [review]

File: src/components/modal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+import * as Styles from './styles';
+
+const Modal = ({
+  isOpen,
+  onClose,
+  children,
+  hideButtonClose = false,
+}: ChildrenProps<IPopup>) => {
+  const [closing, setClosing] = useState(false);
+  const duration = 1000;
```
Comment (Wolshebnik) [inline src/components/modal/index.tsx:16]: duration should be 300

**Review by Kyslytsia — APPROVED** [review]

File: src/components/modal/styles.ts [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+  100% {
+    opacity: 1;
+  }
+`;
+
+const fadeOut = keyframes`
+ 0% {
+    opacity: 1;
+  }
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:18]: space

File: src/components/modal/styles.ts [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+  100%{
+    opacity: 0;
+  }
+`;
+
+const continueMoveDown = keyframes`
+ 0% {
+   opacity: 1;
+   transform: translateY(0vh);
+  }
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:28]: space

File: src/components/modal/styles.ts [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+    opacity: 0;
+     transform: translateY(100vh);
+  }
+`;
+
+const moveDown = keyframes`
+ 0% {
+   opacity: 0;
+     
+  }
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:39]: space

File: src/components/modal/styles.ts [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+  5%{
+    transform: translateY(-100vh);
+  }
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:42]: space

File: src/components/modal/styles.ts [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+  justify-content: center;
+  width: 100vw;
+  height: 100vh;
+  opacity: 1;
+  transition: all 100ms cubic-bezier(0.4, 0, 0.2, 1);
+  z-index: 999;
+
+  ${({ theme, duration, closing }) => css`
+    background-color: ${theme.colors.modalColors.background};
+    animation: ${closing ? appearance : fadeOut} ${duration};
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:64]: ${duration}ms
you forgot

File: src/components/modal/styles.ts [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+  &:after,
+  &:before {
+    position: absolute;
+    top: 7px;
+    content: '';
+    width: 2px;
+    height: 15px;
+    transform: rotate(45deg);
+    background-color: ${({ theme }) => theme.colors.white};
+  }
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:107]: space

File: src/components/modal/styles.ts [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+  &:before {
+    transform: rotate(45deg);
+  }
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:110]: space

File: src/components/modal/styles.ts [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+  &:after {
+    transform: rotate(-45deg);
+  }
```
Comment (Wolshebnik) [inline src/components/modal/styles.ts:113]: space

File: src/components/modal/index.tsx [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+    <>
+      {isOpen && (
+        <Portal>
+          <Styles.Overlay
+            onClick={() => onClose(false)}
+            duration={duration}
+            closing={closing}
+          >
+            <Styles.ModalWindow
+              onClick={(event) => event.stopPropagation()}
```
Comment (Wolshebnik) [inline src/components/modal/index.tsx:44]: sort

**Review by cloudmenson — APPROVED** [review]

File: src/components/modal/index.tsx [hunk 18]
```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+
+import { ChildrenProps } from 'types';
+
+import { IPopup } from './types';
```
Comment (Wolshebnik) [inline src/components/modal/index.tsx:5]: Maybe should rename to IModal

File: src/components/modal/index.tsx [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+      clearTimeout(closeTimeout);
+    }, duration);
+  };
+
+  return (
+    <>
+      {isOpen && (
+        <Portal>
+          <Styles.Overlay
+            onClick={() => onClose(false)}
```
Comment (Wolshebnik) [inline src/components/modal/index.tsx:39]: sort

**Review by ASGladchenko — APPROVED** [review]

**Pull request merged**